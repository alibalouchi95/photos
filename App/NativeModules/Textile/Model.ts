// export type BlockId = string & { _blockIdBrand: void }

export interface File {
  readonly mill: string
  readonly checksum: string
  readonly source: string
  readonly opts?: string
  readonly hash: string
  readonly key?: string
  readonly media: string
  readonly name?: string
  readonly size: number
  readonly added: string
  readonly meta?: ReadonlyMap<string, any>
  readonly targets?: ReadonlyArray<string>
}

export interface Directory {
  readonly [key: string]: File
}

export interface ThreadFileInfo {
  readonly index: number
  readonly file?: File
  readonly links?: Directory
}

export interface ThreadCommentInfo {
  readonly id: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly body: string
}

export interface ThreadLikeInfo {
  readonly id: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
}

export interface ThreadFilesInfo {
  readonly block: string
  readonly target: string
  readonly date: string
  readonly author_id: string
  readonly username?: string
  readonly caption?: string
  readonly files: ReadonlyArray<ThreadFileInfo>
  readonly comments: ReadonlyArray<ThreadCommentInfo>
  readonly likes: ReadonlyArray<ThreadLikeInfo>
  readonly threads: ReadonlyArray<string>
}

export interface BlockInfo {
  readonly id: string
  readonly thread_id: string
  readonly author_id: string
  readonly username: string
  readonly type: string // TODO: No more enum?
  readonly date: string
  readonly parents: ReadonlyArray<string>
  readonly target?: string
  readonly body?: string
}

export interface ExternalInvite {
  readonly id: string
  readonly key: string
  readonly inviter: string
}

export interface CafeSession {
  readonly cafe_id: string
  readonly access: string
  readonly refresh: string
  readonly expiry: string
  readonly http_addr: string
  readonly swarm_addrs: ReadonlyArray<string>
}

export interface Contact {
  readonly id: string
  readonly username: string
  readonly thread_ids: ReadonlyArray<string>
  readonly added: string
}

export enum NotificationType {
  InviteReceivedNotification,
  AccountPeerAddedNotification,
  PeerJoinedNotification,
  PeerLeftNotification,
  MessageAddedNotification,
  FilesAddedNotification,
  CommentAddedNotification,
  LikeAddedNotification
}

export interface Notification {
  readonly id: string
  readonly date: string
  readonly actor_id: string
  readonly subject: string
  readonly subject_id: string
  readonly block_id?: string
  readonly target?: string
  readonly type: NotificationType
  readonly body: string
  readonly read: boolean
}

export interface Overview {
  readonly swarm_size: number
  readonly account_peer_count: number
  readonly thread_count: number
  readonly photo_count: number
  readonly contact_count: number
}

export interface Profile {
  readonly address: string
  readonly inboxes?: ReadonlyArray<string>
  readonly username?: string
  readonly avatar_uri?: string
}

export interface FileData {
  readonly url: string
}

export interface Link {
  readonly use?: string
  readonly pin: boolean
  readonly plaintext: boolean
  readonly mill?: string
  readonly opts?: { readonly [key: string]: string }
  readonly json_schema?: { readonly [key: string]: any }
}

export interface Node {
  readonly name?: string
  readonly pin: boolean
  readonly plaintext: boolean
  readonly mill?: string
  readonly opts?: { readonly [key: string]: string }
  readonly json_schema?: { readonly [key: string]: any }
  readonly links?: { readonly [key: string]: Link }
}

// TODO: Use these if we get enums on ThreadInfo
export enum ThreadType {
  PrivateThread,
  ReadOnlyThread,
  PublicThread,
  OpenThread
}

export enum ThreadState {
  ThreadLoading,
  ThreadLoaded
}

export interface ThreadInfo {
  readonly id: string
  readonly key: string
  readonly name: string
  readonly schema?: Node
  readonly schem_id?: string
  readonly initiator: string
  readonly type: string // TODO: no enum?
  readonly state: string // TODO: no enum?
  readonly head?: BlockInfo
  readonly peer_cnt: number
  readonly block_cnt: number
  readonly file_cnt: number
}

export interface WalletAccount {
  readonly Seed: string // TODO: no json name mappings for these in textile-go
  readonly Address: string
}

export interface ThreadUpdate {
  block: BlockInfo
  thread_id: string
  thread_name: string
  info?: { [key: string]: any } // interface{} is this correct?
}

export enum UpdateType {
  ThreadAdded,
  ThreadRemoved,
  AccountPeerAdded,
  AccountPeerRemoved
}

export interface Update {
  id: string
  name: string
  type: UpdateType
}
