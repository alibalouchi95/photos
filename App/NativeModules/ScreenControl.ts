import { NativeModules } from 'react-native'

const { ScreenControl } = NativeModules

export function letScreenSleep() {
  ScreenControl.letScreenSleep()
}

export function keepScreenOn() {
  ScreenControl.keepScreenOn()
}
