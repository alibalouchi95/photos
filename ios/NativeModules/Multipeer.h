#import <Foundation/Foundation.h>
//#import <React/RCTBridgeModule.h>
#import <MultipeerConnectivity/MultipeerConnectivity.h>
#import "PeerBrowser.h"

#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#elif __has_include("RCTBridgeModule.h")
#import “RCTBridgeModule.h”
#else
#import "React/RCTBridgeModule.h"
#endif
#import <React/RCTBridge.h>

@interface Multipeer : NSObject <MCSessionDelegate, RCTBridgeModule>

@property (nonatomic, strong) NSString *name;
@property (nonatomic, strong) NSString *address;
@property (nonatomic, strong) NSString *avatar;
@property (nonatomic, strong) MCPeerID *peerID;
@property (nonatomic, strong) MCSession *session;
//@property (nonatomic, strong) MCBrowserViewController *browser;
@property (nonatomic, strong) PeerBrowser *browser;
@property (nonatomic, strong) MCAdvertiserAssistant *advertiser;

- (void)initialize:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar;
- (void)advertiseToggle:(BOOL)onOff;
- (void)scanStart;

@end

@interface RCTBridge (Multipeer)

@property (nonatomic, readonly) Multipeer *multipeer;

@end
