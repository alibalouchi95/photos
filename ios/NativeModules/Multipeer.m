#import "Multipeer.h"

@interface Multipeer()

@end

@implementation Multipeer

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(init:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar) {
  [self initialize:name address:address avatar:avatar];
}

RCT_EXPORT_METHOD(advertiseOn) {
  [self advertiseToggle:true];
}

RCT_EXPORT_METHOD(advertiseOff) {
  [self advertiseToggle:false];
}

RCT_EXPORT_METHOD(scan) {
  [self scanStart];
}

#pragma mark -
#pragma mark Public Methods
- (void)initialize:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar {
  self.name = name;
  self.address = address;
  self.avatar = avatar;
  self.peerID = [[MCPeerID alloc] initWithDisplayName:address];
  self.session = [[MCSession alloc] initWithPeer:self.peerID];
  self.session.delegate = self;
}

- (void)scanStart {
  self.browser = [[PeerBrowser alloc] init];
  [self.browser initialize:@"textileio" peerID:self.peerID];
}

- (void)advertiseToggle:(BOOL)advertise {
  if (advertise) {
    NSDictionary *discoveryInfo = @{ @"address": self.address, @"avatar": self.avatar, @"name": self.name };
    self.advertiser = [[MCAdvertiserAssistant alloc] initWithServiceType:@"textileio" discoveryInfo:discoveryInfo session:self.session];
    [self.advertiser start];
  } else {
    [self.advertiser stop];
    self.advertiser = nil;
  }
}

#pragma mark -
#pragma mark Session Delegate Methods
- (void)session:(MCSession *)session peer:(MCPeerID *)peerID didChangeState:(MCSessionState)state {
//  NSDictionary *userInfo = @{ @"peerID": peerID,
//                              @"state" : @(state) };
//  dispatch_async(dispatch_get_main_queue(), ^{
//    [[NSNotificationCenter defaultCenter] postNotificationName:@"Multipeer_DidChangeStateNotification"
//                                                        object:nil
//                                                      userInfo:userInfo];
//  });
}

- (void)session:(MCSession *)session didReceiveData:(NSData *)data fromPeer:(MCPeerID *)peerID {
//  NSDictionary *userInfo = @{ @"data": data,
//                              @"peerID": peerID };
//  dispatch_async(dispatch_get_main_queue(), ^{
//    [[NSNotificationCenter defaultCenter] postNotificationName:@"Multipeer_DidReceiveDataNotification"
//                                                        object:nil
//                                                      userInfo:userInfo];
//  });
}

- (void)session:(MCSession *)session didStartReceivingResourceWithName:(NSString *)resourceName fromPeer:(MCPeerID *)peerID withProgress:(NSProgress *)progress {

}

- (void)session:(MCSession *)session didFinishReceivingResourceWithName:(NSString *)resourceName fromPeer:(MCPeerID *)peerID atURL:(NSURL *)localURL withError:(NSError *)error {

}

- (void)session:(MCSession *)session didReceiveStream:(NSInputStream *)stream withName:(NSString *)streamName fromPeer:(MCPeerID *)peerID {

}

@end
