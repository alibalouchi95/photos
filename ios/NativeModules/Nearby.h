#import <Foundation/Foundation.h>
//#import <React/RCTBridgeModule.h>
#import <CoreBluetooth/CoreBluetooth.h>

#import "Peripheral.h"

#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#elif __has_include("RCTBridgeModule.h")
#import “RCTBridgeModule.h”
#else
#import "React/RCTBridgeModule.h"
#endif
#import <React/RCTBridge.h>

@interface Nearby : NSObject <CBCentralManagerDelegate, RCTBridgeModule>

@property (nonatomic, strong) NSString *name;
@property (nonatomic, strong) NSString *address;
@property (nonatomic, strong) NSString *avatar;
@property (nonatomic, strong) CBUUID *appUUID;
// CBCentral properties
@property (nonatomic, strong) NSMutableData *finaldata;
@property (nonatomic, strong) Peripheral *peripheral;
@property (nonatomic, strong) CBCentralManager *manager;
//@property (nonatomic, strong) CBMutableCharacteristic *characteristic;

- (void)initialize:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar;
- (void)initScan;

@end

@interface RCTBridge (Nearby)

@property (nonatomic, readonly) Nearby *nearby;

@end
