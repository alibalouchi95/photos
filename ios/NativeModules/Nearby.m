#import "Nearby.h"

@interface Nearby()

@end

@implementation Nearby

RCT_EXPORT_MODULE();

RCT_EXPORT_METHOD(init:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar) {
  [self initialize:name address:address avatar:avatar];
}

RCT_EXPORT_METHOD(scan) {
  [self initScan];
}

RCT_EXPORT_METHOD(advertise) {
  [self startAdvertising];
}
//
//RCT_EXPORT_METHOD(advertiseOff) {
//  [self advertiseToggle:false];
//}

#pragma mark -
#pragma mark Public Methods
- (void)initialize:(NSString *)name address:(NSString *)address avatar:(NSString *)avatar {
  self.name = name;
  self.address = address;
  self.avatar = avatar;
  self.appUUID = [CBUUID UUIDWithString:@"0000b81d-0000-1000-8000-00805f8b34fb"];

  self.finaldata = [[NSMutableData alloc]init];
  self.manager = [[CBCentralManager alloc] init];
  self.manager.delegate = self;
  self.peripheral = [[Peripheral alloc] init];
//  self.peripheral.initialize(name, address, avatar);
  [self.peripheral initialize:name address:address avatar:avatar];
}

- (void)startAdvertising {
//  self.periphera.initialize
  //l = [[Peripheral alloc] initialize:name address:address avatar:avatar];
  NSDictionary *discoveryInfo = @{ @"address": self.address, @"avatar": self.avatar, @"name": self.name };
  [self.peripheral advertise:discoveryInfo];
}

- (void) initScan {
  [self.manager stopScan];
  //  Look for services only with our shared app uuid
  NSArray *services = [NSArray arrayWithObjects:self.appUUID, nil];
  NSDictionary *discoveryInfo = @{ @"address": self.address, @"avatar": self.avatar, @"name": self.name };
  [self.manager
   scanForPeripheralsWithServices:nil
   options:nil
//   options:[NSDictionary dictionaryWithObjectsAndKeys:discoveryInfo, CBCentralManagerScanOptionAllowDuplicatesKey, nil]
  ];
}

- (void) peripheral:(CBPeripheral *)aPeripheral didDiscoverServices:(nullable NSError *)error {

  NSLog(@"Yoo");
}

- (void) peripheral:(CBPeripheral *)aPeripheral didDiscoverCharacteristicsForService:(CBService *)service error:(NSError *)error
{

  NSLog(@"HEYOOOO");
  NSLog(@"%@",aPeripheral.identifier);
  NSLog(@"%@",aPeripheral.observationInfo);
  NSLog(@"%@",aPeripheral.description);
  NSLog(@"more?");
  for (CBCharacteristic *aChar in service.characteristics){
    NSLog(@"%@",aChar.UUID);
//    [TextView insertText:[NSString stringWithFormat:@"Characteristic UUID:%@\n",aChar.UUID]];
    if ([aChar.UUID isEqual:[CBUUID UUIDWithString:@"DA18"]]) {
      NSLog(@"%lu",aChar.properties);
//      [TextView insertText:[NSString stringWithFormat:@"Characteristic Prop:%lu\n",aChar.properties]];
      [aPeripheral setNotifyValue:YES forCharacteristic:aChar];
    }
    if ([aChar.UUID isEqual:[CBUUID UUIDWithString:@"DA17"]]) {
      //NSLog(@"Find DA17");
      NSLog(@"%lu",aChar.properties);
//      [TextView insertText:[NSString stringWithFormat:@"Characteristic Prop:%lu\n",aChar.properties]];
      NSString *mainString = [NSString stringWithFormat:@"12345"];
      NSData *mainData= [mainString dataUsingEncoding:NSUTF8StringEncoding];
      [aPeripheral writeValue:mainData forCharacteristic:aChar type:CBCharacteristicWriteWithResponse];
    }
    if ([aChar.UUID isEqual:[CBUUID UUIDWithString:@"DA16"]]) {
      NSLog(@"Find DA16");
      NSLog(@"%lu",aChar.properties);
//      [TextView insertText:[NSString stringWithFormat:@"Characteristic Prop:%lu\n",aChar.properties]];
      //[aPeripheral readValueForCharacteristic:aChar];
    }
  }
}

- (void)peripheral:(CBPeripheral *)peripheral didWriteValueForCharacteristic:(CBCharacteristic *)characteristic error:(NSError *)error{
  NSLog(@"Finish Write\n");
//  [TextView insertText:@"Finish Write\n"];
}

- (void) peripheral:(CBPeripheral *)aPeripheral didUpdateValueForCharacteristic:(CBCharacteristic *)characteristic error:(NSError *)error
{
  NSData * updatedValue = characteristic.value;
  NSLog(@"%@",[[NSString alloc]initWithData:updatedValue encoding:NSUTF8StringEncoding]);
  if ([[[NSString alloc]initWithData:updatedValue encoding:NSUTF8StringEncoding]isEqualToString:@"ENDAL"]) {
    [self.manager cancelPeripheralConnection:aPeripheral];
//    [TextView insertText:[NSString stringWithFormat:@"%@\n",[[NSJSONSerialization JSONObjectWithData:finaldata options:kNilOptions error:nil]description]]];
  }else{
    [self.finaldata appendData:updatedValue];
  }
}

- (IBAction)Connect:(id)sender {
  //A workaround for CBPeripheralManager
  [self.manager scanForPeripheralsWithServices:@[ self.appUUID ] options:@{ CBCentralManagerScanOptionAllowDuplicatesKey : [NSNumber numberWithBool:YES] }];
//  Button.title = @"Scanning";
//  [Button setEnabled:NO];
}

- (IBAction)disconnect:(id)sender {
//  [self.manager cancelPeripheralConnection:self.peripheral];
}

- (void)centralManagerDidUpdateState:(nonnull CBCentralManager *)central {
  NSLog(@"We're something");
  if (central.state == CBManagerStatePoweredOn) {
    NSLog(@"We're COOKING");

//    CBCharacteristicProperties props = CBCharacteristicPropertyBroadcast;
//    CBAttributePermissions perms = CBAttributePermissionsReadable;
//    CBMutableCharacteristic *characteristic = [[CBMutableCharacteristic alloc] initWithType:self.appUUID properties:props value:NULL permissions:perms];
//    CBMutableService *service = [[CBMutableService alloc] initWithType:self.appUUID primary:TRUE];

  }
  NSLog(@"%ld",[self.manager state]);
}

- (void)centralManager:(CBCentralManager *)central didDiscoverPeripheral:(CBPeripheral *)peripheral advertisementData:(NSDictionary<NSString *,id> *)advertisementData RSSI:(NSNumber *)RSSI {
  NSLog(@"AXH");

  NSLog( @"%@", advertisementData);
  NSLog( @"%@", peripheral.services);
  NSLog( @"%@", peripheral.description);
  NSString *name = peripheral.name ? peripheral.name : @"no name";
  if ([name isEqualToString:@"Pixel 3"]) {
    NSLog( @"AXH %@", @"Jackpot" );
    [self.manager connectPeripheral:peripheral options:nil];
    NSLog( @"AXH %@", @"Jackpot" );
    [peripheral discoverServices:nil];
  }
  NSLog(@"AXH %@", name);
}

@end
