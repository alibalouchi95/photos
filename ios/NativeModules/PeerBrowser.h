#import <Foundation/Foundation.h>
//#import <React/RCTBridgeModule.h>
#import <MultipeerConnectivity/MultipeerConnectivity.h>

#if __has_include(<React/RCTBridgeModule.h>)
#import <React/RCTBridgeModule.h>
#elif __has_include("RCTBridgeModule.h")
#import “RCTBridgeModule.h”
#else
#import "React/RCTBridgeModule.h"
#endif
#import <React/RCTBridge.h>

@interface PeerBrowser : NSObject <MCNearbyServiceBrowserDelegate, RCTBridgeModule>

@property (nonatomic, strong) MCNearbyServiceBrowser *browser;
//@property (nonatomic, strong) MCPeerID *peerID;
//@property (nonatomic, strong) MCSession *session;
//@property (nonatomic, strong) MCBrowserViewController *browser;
//@property (nonatomic, strong) MCNearbyServiceBrowser *browser;
//@property (nonatomic, strong) MCAdvertiserAssistant *advertiser;

- (void)initialize:(NSString *)serviceType peerID:(MCPeerID *)peerID;


@end

@interface RCTBridge (PeerBrowser)

@property (nonatomic, readonly) PeerBrowser *peerBrowser;

@end
