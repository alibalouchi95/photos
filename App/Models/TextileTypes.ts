// BEGIN API Types

// private enums to enforce strict id types
// read here: https://basarat.gitbooks.io/typescript/docs/tips/nominalTyping.html#using-enums

import {IProcessingImageProps} from '../Components/ProcessingImage'
import { ThreadFilesInfo } from '../NativeModules/Textile'

export interface IPhotoGridType {
  type: 'photo' | 'processingItem'
  photo: ThreadFilesInfo | IProcessingImageProps
  id: string
}

export interface SharedImage {
  origURL?: string,
  uri: string,
  path: string,
  canDelete: boolean
}

export interface DeepLinkData {
  readonly href: string,
  readonly protocol: string,
  readonly host: string,
  readonly hostname: string,
  readonly port: string,
  readonly path: string,
  readonly search: string
  readonly hash: string
}

export interface ILocalPhotoResult {
  assetId: string,
  creationDate: string,
  modificationDate: string,
  orientation: number,
  path: string,
  uri: string,
  canDelete: boolean
}
