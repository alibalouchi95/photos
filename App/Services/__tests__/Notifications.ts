import {
    enable,
    toPayload
} from '../Notifications'
import { NotificationType } from '../../NativeModules/Textile'
import { Notification } from '../../Models/Notifications'

describe('notifications', () => {
    describe('enable', () => {
        it('should fire notification lib successfully', async () => {
            await expect(enable()).resolves.toMatchSnapshot()
        })
    })
    describe('toPayload', () => {
        const base = {
            id: 'noteId',
            date: '2018-12-11T22:34:26.265Z',
            actor_id: 'EFG',
            read: false,
            username: 'mock test',
            threadId: 'ABC',
            threadName: 'Great Thread',
            target: 'DEF'
        }
        it('to create InviteReceivedNotification', () => {
            const notification = {
                ...base,
                body: 'invited you to join',
                type: NotificationType.InviteReceivedNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
        it('to create AccountPeerJoinedNotification', () => {
            const notification = {
                ...base,
                body: 'joined',
                type: NotificationType.AccountPeerJoinedNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
        it('to create PeerJoinedNotification', () => {
            const notification = {
                ...base,
                body: 'joined',
                type: NotificationType.PeerJoinedNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
        it('to create PeerLeftNotification', () => {
            const notification = {
                ...base,
                body: 'left',
                type: NotificationType.PeerLeftNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
        it('to create MessageAddedNotification', () => {
            const notification = {
                ...base,
                body: 'here is my message!',
                type: NotificationType.MessageAddedNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
        it('to create FilesAddedNotification', () => {
            const notification = {
                ...base,
                body: 'added a photo',
                type: NotificationType.FilesAddedNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
        it('to create CommentAddedNotification', () => {
            const notification = {
                ...base,
                body: 'commented on a photo: a camel!',
                type: NotificationType.CommentAddedNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
        it('to create LikeAddedNotification', () => {
            const notification = {
                ...base,
                body: 'liked a photo',
                type: NotificationType.LikeAddedNotification
            }
            expect(toPayload(notification as Notification)).toMatchSnapshot()
        })
    })
})
