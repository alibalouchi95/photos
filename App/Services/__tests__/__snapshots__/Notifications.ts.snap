// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notifications enable should fire notification lib successfully 1`] = `undefined`;

exports[`notifications toPayload to create account peer notification correctly 1`] = `
Object {
  "feed": "You paired with a new account peer",
  "message": "You paired with a new account peer",
  "title": "New Account Peer",
  "typeString": "ACCOUNT_PEER_JOINED",
}
`;

exports[`notifications toPayload to create comment added notification correctly 1`] = `
Object {
  "feed": "mock test commented on a photo in Great Thread",
  "message": "mock test commented on a photo: a camel!",
  "title": "Great Thread",
  "typeString": "COMMENT_ADDED",
}
`;

exports[`notifications toPayload to create invite notification correctly 1`] = `
Object {
  "feed": "mock test invited you to join to Great Thread",
  "message": "mock test invited you to join",
  "title": "New Invite",
  "typeString": "INVITE_RECEIVED",
}
`;

exports[`notifications toPayload to create messaged added notification correctly 1`] = `
Object {
  "feed": "mock test here is my message! to Great Thread",
  "message": "mock test here is my message!",
  "title": "Great Thread",
  "typeString": "MESSAGE_ADDED",
}
`;

exports[`notifications toPayload to create new like notification correctly 1`] = `
Object {
  "feed": "mock test liked a photo in Great Thread",
  "message": "mock test liked a photo",
  "title": "Great Thread",
  "typeString": "LIKE_ADDED",
}
`;

exports[`notifications toPayload to create peer joined notification correctly 1`] = `
Object {
  "feed": "mock test joined thread Great Thread",
  "message": "mock test joined",
  "title": "Great Thread",
  "typeString": "PEER_JOINED",
}
`;

exports[`notifications toPayload to create peer left notification correctly 1`] = `
Object {
  "feed": "mock test left thread Great Thread",
  "message": "mock test left",
  "title": "Great Thread",
  "typeString": "PEER_LEFT",
}
`;

exports[`notifications toPayload to create photo notification correctly 1`] = `
Object {
  "feed": "mock test added a photo to Great Thread",
  "message": "mock test added a photo",
  "title": "Great Thread",
  "typeString": "FILES_ADDED",
}
`;
